File: critter_storage/premade/Bear.java
package critter_storage.premade;
import java.awt.*;
import game.*;

public class Bear extends Critter { 
    boolean polar = true;
    boolean isOdd = true;

  
    public Bear (boolean polar) {
        this.polar = polar;
    }

    public Action getMove(CritterInfo info) {
        if (info.getFront() == Neighbor.OTHER) {
            return Action.INFECT;
        } else if (info.getFront() == Neighbor.EMPTY){
            return Action.HOP;
        }else{
            return Action.LEFT;
            
        }
    }
   public String getName(){
       return "Bear";
   }
    public Color getColor() {
        
          if(polar){          
            return Color.WHITE;
          }else{
            return Color.BLACK;
          }       
    }
        
    public String toString() {
        if(isOdd){ 
            isOdd = false;
            return "/";
        }else{
            isOdd = true;
            return "\\";
        }
    }
}

File: CSharpVersion/CritterStorage/Premade/Bear.cs
using System.Drawing;
using CritterSimulator.Game;

namespace CritterSimulator.CritterStorage.Premade
{
    public class Bear : Critter
    {
        private bool polar = true;
        private bool isOdd = true;

        public Bear(bool polar)
        {
            this.polar = polar;
        }

        public override Action GetMove(ICritterInfo info)
        {
            if (info.GetFront() == Neighbor.Other)
            {
                return Action.Infect;
            }
            else if (info.GetFront() == Neighbor.Empty)
            {
                return Action.Hop;
            }
            else
            {
                return Action.Left;
            }
        }

        public override Color GetColor()
        {
            if (polar)
            {
                return Color.White;
            }
            else
            {
                return Color.Black;
            }
        }

        public override string ToString()
        {
            if (isOdd)
            {
                isOdd = false;
                return "/";
            }
            else
            {
                isOdd = true;
                return "\\";
            }
        }
    }
}
